File: src/main/java/recipeapplication/model/ShoppingListItem.java
package recipeapplication.model;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "shopping_list")
public class ShoppingListItem implements Serializable
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    private User user;

    @OneToOne
    private Ingredient ingredient;

    public Long getId()
    {
        return id;
    }

    public User getUser()
    {
        return user;
    }

    public void setUser(User user)
    {
        this.user = user;
    }

    public Ingredient getIngredient()
    {
        return ingredient;
    }

    public void setIngredient(Ingredient ingredient)
    {
        this.ingredient = ingredient;
    }
}

File: src/main/kotlin/thecookingpot/recipe/model/ShoppingListItem.kt
package thecookingpot.recipe.model

import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "shopping_list")
class ShoppingListItem : Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    @OneToOne
    lateinit var user: User

    @OneToOne(cascade = [CascadeType.PERSIST])
    lateinit var ingredient: Ingredient
}